hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieIndexConfig.java

```diff
@@ -334,6 +334,11 @@ public class HoodieIndexConfig extends HoodieConfig {
       .withDocumentation("Only applies when #recordIndexUseCaching is set. Determine what level of persistence is used to cache input RDDs. "
           + "Refer to org.apache.spark.storage.StorageLevel for different values");
 
+  public static final ConfigProperty<Boolean> BUCKET_QUERY_INDEX = ConfigProperty
+      .key("hoodie.bucket.query.index")
```

Comment:
Can we eliminate this option? The improvement should always be atomatic.

Reply:
bucket query index will auto infer, this conf is give a choice when there is bug or special business scenario user can close this infer.

special business scenario: user change the  bucket number but didn't rewrite history partitions, because history data will not update in there scenario, this case can not use bucket index query

Reply:
I don't think we should support changing the bucket numer for a table, user must do a rewrite.

Reply:
`hoodie.bucket.query.index` -> `hoodie.bucket.index.query.pruning` ?